### PR TITLE
AudioPlayer

### PR DESCRIPTION
**Pull request by timdubbins**
I found a few bugs with the player:
- ‘status’ property does not change to stopped when the player completes playback
- unable to stop a paused file / buffer 
- getCurrentTime() doesn’t show correct time if the same file / buffer completes playback and is then played again
- getCurrentTime() does not show the correct time when looping

These should be fixed now. I wasn’t sure if  changing ‘getCurrentTime()’ to ‘currentTime’ was a good idea or if that sort of name-changing is annoying but included it to find out.

Also noticed that ‘seek(time:)’  is not working. I should be able to fix this as I have a working seek function in the app I built, will take a look at it.

### CODE REVIEW

**Review by emurray2 — APPROVED** [review]

**Comment by emurray2** [comment]
The changes look great! Do you want to add in some more test cases for the `status` and stopping the paused buffer/file, as well as add your two test cases under `testCurrentTime()`?

**Comment by timdubbins** [comment]
Thanks! I'll take a look at seek and write some tests

**Comment by timdubbins** [comment]
Need to add the tests but think currentTime and seek should be working now. Seek works like a "skip-forward" or "skip-backward" button, so you can just increment your way through a file. Can also be used as a "fast-forward" or "rewind" button if used in conjunction with a long-press gesture

File: Sources/AudioKit/Nodes/Playback/AudioPlayer/AudioPlayer+Buffering.swift [hunk 1]
```diff
@@ -5,7 +5,7 @@ import AVFoundation
 /// Functions specific to buffering audio
 extension AudioPlayer {
     // Fills the buffer with data read from the linked audio file
-    func updateBuffer(force: Bool = false) {
+    func updateBuffer() {
```
[the remark this answers is not in the document]
Comment (timdubbins) [reply Sources/AudioKit/Nodes/Playback/AudioPlayer/AudioPlayer+Buffering.swift:8]: I'm not really sure how to fix this hound message

**Comment by timdubbins** [comment]
I noticed in the test comments that it might not be possible to test the state of the player that is changed from the completion handler (it mentions that the callback happens on a different thread). Is there any way around this?

File: Sources/AudioKit/Nodes/Playback/AudioPlayer/AudioPlayer+Buffering.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (timdubbins) [reply Sources/AudioKit/Nodes/Playback/AudioPlayer/AudioPlayer+Buffering.swift:8]: The whitespace doesn't seem to exist in my file...

**Review by aure — APPROVED** [review]

**Comment by emurray2** [comment]
> I noticed in the test comments that it might not be possible to test the state of the player that is changed from the completion handler (it mentions that the callback happens on a different thread). Is there any way around this?

We could probably have a debug mode of the player where it will dispatch the status back to the tests somehow, but this test suite is part of the realtime tests anyway which aren't enabled in CI at the moment. You're welcome to add something, but I think it could make sense for another PR since this one has a decent amount of commits already. Everything looks good, and thank you for your improvements :D !

**Review by emurray2 — APPROVED** [review]

**Comment by timdubbins** [comment]
Happy to contribute. I will have a look at the realtime tests at some point. I really like the startTest and render methods from AudioEngine, wish I'd found them sooner. There's a lot of good stuff in this repo, it's an education ;)

**Comment by warpling** [comment]
Thank you @timdubbins!! This was driving me nuts!

**Pull request merged**